test/createAzureResource.test.ts

```diff
@@ -59,26 +59,83 @@ suite('Create Azure Resources', async function (this: ISuiteCallbackContext): Pr
         ext.azureAccountTreeItem.dispose();
     });
 
-    test('createFunctionApp (Basic)', async () => {
+    test('Create windows function app (Basic) and deploy JavaScript project', async () => {
+        const functionName: string = 'HttpTrigger';
         resourceGroupsToDelete.push(resourceName1);
-        await runWithFuncSetting('advancedCreation', undefined, async () => {
-            ext.ui = new TestUserInput([resourceName1]);
-            await vscode.commands.executeCommand('azureFunctions.createFunctionApp');
-            const createdApp: WebSiteManagementModels.Site = await webSiteClient.webApps.get(resourceName1, resourceName1);
-            assert.ok(createdApp);
+        await runWithFuncSetting('projectLanguage', ProjectLanguage.JavaScript, async () => {
+            await runWithFuncSetting('advancedCreation', undefined, async () => {
+                ext.ui = new TestUserInput([resourceName1]);
+                await vscode.commands.executeCommand('azureFunctions.createFunctionApp');
+                const createdApp: WebSiteManagementModels.Site = await webSiteClient.webApps.get(resourceName1, resourceName1);
+                assert.ok(createdApp);
+                await runWithFuncSetting('projectRuntime', ProjectRuntime.v2, async () => {
+                    const templateId: string = 'HttpTrigger-JavaScript';
+                    const authLevel: string = 'function';
+                    ext.ui = new TestUserInput([DialogResponses.skipForNow.title]);
+                    await vscode.commands.executeCommand('azureFunctions.createNewProject', projectPath, ProjectLanguage.JavaScript, ProjectRuntime.v2, false, templateId, functionName, { AuthLevel: authLevel });
+                    await validateProject(projectPath, getJavaScriptValidateOptions(true));
+                });
+            });
         });
+
+        ext.ui = new TestUserInput([resourceName1, 'Deploy']);
+        await vscode.commands.executeCommand('azureFunctions.deploy');
+        await delay(500);
+
+        // Verify the deployment result through triggerUrl
+        const getFunction: WebSiteManagementModels.FunctionEnvelopeCollection = await webSiteClient.webApps.listFunctions(resourceName1, resourceName1);
+        assert.equal(getFunction[0].name, `${resourceName1}/${functionName}`);
+        const fullId: string = <string>(getFunction[0].id);
+        const triggerUrl: string | undefined = (<FunctionTreeItem>await ext.tree.findTreeItem(fullId, <IActionContext>context)).triggerUrl;
+        const result: string = execSync(`curl ${triggerUrl}"&"name=${resourceName1}`).toString();
```

Comment:
node.js already has ways to make a request without using a cli tool like `curl`. Take a look at something like this: https://github.com/microsoft/vscode-azurefunctions/blob/master/src/utils/getCliFeedJson.ts#L45